rmf_traffic/include/rmf_traffic/agv/Planner.hpp

```diff
@@ -333,6 +333,37 @@ class Planner
   /// Get a const reference to the default planning options.
   const Options& get_default_options() const;
 
+  using StartSet = std::vector<Start>;
+
+  /// Produces a set of starting conditions in order to start planning. This
+  /// method attempts to find the most suitable starting nodes within the graph
+  /// for merging, planning and execution of plans. If none of the waypoints in
+  /// the graph fulfils the requirements, a nullopt will be returned.
+  ///
+  /// \param[in] pose
+  ///   Current pose in terms of 2D coordinates, x and y, being the first and
+  ///   second element respectively, while the third element being the yaw.
+  ///
+  /// \param[in] start_time
+  ///   The starting time that will be attributed to all the generated starts
+  ///   to compute a new plan. In some occations, users will want to add small
+  ///   delays to the current time, in order to account for computation time or
+  ///   network delays.
+  ///
+  /// \param[in] max_merge_waypoint_distance
+  ///   The maximum distance allowed to automatically merge onto a waypoint in 
+  ///   the graph, default value as 0.1 meters.
+  ///
+  /// \param[in] max_merge_lane_distance
+  ///   The maximum distance allowed to automatically merge onto a lane, i.e.
+  ///   adding the lane's entry and exit waypoints as potential starts. default
+  ///   value as 1.0 meters.
+  rmf_utils::optional<StartSet> compute_plan_starts(
```

Comment:
Would you be open to the idea of having it as something under a new header called `Behaviour`? Probably under `Behaviour::PlanStarter` with `Behaviour::PlanStarter::Options`.

This might be helpful when we start to identify other behaviours we might want AGVs to follow within RMF.

Reply:
just a note about our conversation, `compute_plan_starts` will be migrated to a utility function under `Planner.hpp`, with all the needed parameters passed in